rita_common/src/peer_listener/mod.rs

```diff
@@ -103,36 +145,46 @@ fn listen_to_available_ifaces(peer_listener: &mut PeerListener) {
                         .interfaces
                         .insert(new_listen_interface.ifname.clone(), new_listen_interface);
                 }
-                Err(_e) => {}
+                Err(e) => {
+                    error!("Received an error while listening to interfaces: {:?}", e)
+                }
             }
         }
     }
 }
 
+/// Returns a copy of PL lazy static var
+fn get_pl_copy() -> PeerListener {
+    PEER_LISTENER.read().unwrap().clone()
+}
+
+/// Sets the PeerListener lazy static to the given value
+fn set_pl(pl: PeerListener) {
+    *PEER_LISTENER.write().unwrap() = pl;
+}
+
 /// Ticks the peer listener module sending ImHere messages and receiving Hello messages from all
 /// peers over UDP
-pub fn peerlistener_tick() {
+pub fn peerlistener_tick() -> PeerListener {
     trace!("Starting PeerListener tick!");
 
-    let mut writer = PEER_LISTENER.write().unwrap();
-    if !writer.contacting_neighbors {
-        send_im_here(&mut writer.interfaces);
-
-        let (a, b) = receive_im_here(&mut writer.interfaces);
-        {
-            // Reset hashmaps only when not contacting peers
-            (*writer).peers = a;
-            (*writer).interface_map = b;
-        }
-        receive_hello(&mut writer);
-
-        listen_to_available_ifaces(&mut writer);
+    let mut pl = get_pl_copy();
+    send_im_here(&mut pl.interfaces);
+    let (a, b) = receive_im_here(&mut pl.interfaces);
+    {
+        pl.peers = a;
+        pl.interface_map = b;
     }
+    receive_hello(&mut pl);
+    listen_to_available_ifaces(&mut pl);
+
+    set_pl(pl.clone());
+    pl
```

Comment:
Im assuming this is fine or should I return a cloned udpsocket and save the original?

Reply:
👍🏻